src/bech32.cpp

```diff
@@ -150,6 +150,9 @@ std::string Encode(const std::string& hrp, const data& values) {
     std::string ret = hrp + '1';
     ret.reserve(ret.size() + combined.size());
     for (auto c : combined) {
+        if (c >= 32) {
+            return "";
+        }
```

Comment:
I'm unconvinced that this should be returning "" as an error indicator. It makes more sense for it to be a precondition that ``values`` does not contain out-of-range values; failure of the precondition would then be reported via an assertion failure or exception.

Reply:
If we change `bech32::Encode` to not return `""` on encoding error, then we should also change `bech32::Decode` to not use an empty `hrp` to indicate a decoding error, in order to be consistent. That would make the decode logic a bit more complex, as we'd need to separately handle the error case from the wrong-HRP case (whereas currently they are handled together). But would also be technically clearer, so I'm ambivalent.

Reply:
I guess it's fine, but it should certainly be better-documented (here and for ``Decode``).